### PR TITLE
XML Sitemap API docs (and some meta descriptions)

### PR DESCRIPTION
**Pull request by jono-alderson**
- Added content to the XML sitemaps doc page
- Added meta descriptions to existing API pages

@jcomack please could you review the 'coming soon' sections and provide examples?

### CODE REVIEW

File: docs/features/xml-sitemaps/api.md [hunk 1]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
 description: Coming soon.
 ---
 
-Coming soon.
+Whilst Yoast SEO provides sensible default behaviours for XML sitemaps (and UI controls for inclusion/exclusion), custom themes or plugins sometimes need to alter our markup or logic.
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:9]: `behaviours` needs to be changed. Should just American English.

File: docs/features/xml-sitemaps/api.md [hunk 2]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+## Excluding content types
+
+### Exclude specific posts
+```php
+/**
+ * Excludes posts from XML sitemaps
+ */
+function exclude_posts_from_xml_sitemaps() {
+    $posts = array( 1, 2, 3 );
+    return $posts;
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:21]: I'd suggest just using:

```php
return [ 1, 2, 3 ];
```

File: docs/features/xml-sitemaps/api.md [hunk 3]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
-Coming soon.
+Whilst Yoast SEO provides sensible default behaviours for XML sitemaps (and UI controls for inclusion/exclusion), custom themes or plugins sometimes need to alter our markup or logic.
+In those cases, you can use the examples below to modify how our sitemaps are generated and output.
+
+## Excluding content types
+
+### Exclude specific posts
+```php
+/**
+ * Excludes posts from XML sitemaps
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:17]: We want too ensure our code examples follow our own coding standards, so please add the `@return` property with a description, to the docblock.

File: docs/features/xml-sitemaps/api.md [hunk 4]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+    $posts = array( 1, 2, 3 );
+    return $posts;
+}
+add_filter( 'wpseo_exclude_from_sitemap_by_post_ids', 'exclude_posts_from_xml_sitemaps' );
+```
+
+### Exclude a post type
+```php
+/**
+ * Exclude a post type from XML sitemaps
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:29]: We want too ensure our code examples follow our own coding standards, so please add the parameters and `@return` property with a description, to the docblock.

File: docs/features/xml-sitemaps/api.md [hunk 5]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+ */
+function sitemap_exclude_post_type( $value, $post_type ) {
+    $post_type_slug = 'recipes'; // An example custom post type slug.
+    if ( $post_type == $post_type_slug ) return true;
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:33]: I'd simplify this to:

```php
return $post_type === 'recipes';
```

File: docs/features/xml-sitemaps/api.md [hunk 6]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+}
+add_filter( 'wpseo_sitemap_exclude_post_type', 'sitemap_exclude_post_type', 10, 2 );
+```
+
+### Exclude a taxonomy
+```php
+/**
+ * Exclude a taxonomy from XML sitemaps
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:41]: We want too ensure our code examples follow our own coding standards, so please add the parameters and `@return` property with a description, to the docblock.

File: docs/features/xml-sitemaps/api.md [hunk 7]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+ */
+function sitemap_exclude_taxonomy( $value, $taxonomy ) {
+    $taxonomy_slug = 'ingredients'; // An example taxonomy slug.
+    if ( $taxonomy == $taxonomy_slug ) return true;
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:45]: I'd simplify this to:

```php
return $taxonomy === 'ingredients';
```

File: docs/features/xml-sitemaps/api.md [hunk 8]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+        $wpseo_sitemaps->register_sitemap( $post_type, 'create_TYPE_sitemap' );
+    }
+}
+add_action( 'init', 'enable_custom_sitemap' );
+```
+
+### Add additional/external XML sitemaps to the XML sitemap index
+```php
+/**
+ * Writes additional/custom XML sitemap strings to the XML sitemap index
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:76]: We want too ensure our code examples follow our own coding standards, so please add the parameter and the `@return` property with a description, to the docblock.

File: docs/features/xml-sitemaps/api.md [hunk 9]
```diff
@@ -6,4 +6,103 @@ custom_edit_url: https://github.com/Yoast/developer-docs/edit/master/docs/featur
...
+
+## Misc
+
+## Alter the URL of an entry
+Coming soon. See `wpseo_xml_sitemap_post_url`.
+
+### Alter the number of sitemap entries
+```php
+/**
+ * Alters the number of entries in each XML sitemap
```
Comment (jcomack) [inline docs/features/xml-sitemaps/api.md:101]: We want too ensure our code examples follow our own coding standards, so please add the `@return` property with a description, to the docblock.

**Comment by jono-alderson** [comment]
All done.

**Pull request merged**